internal/discovery/foreign.go

```diff
@@ -132,11 +140,13 @@ func (discovery *DiscoveryCtrl) createForeign(data *discoveryData, trustMode dis
 				UID:        sd.UID,
 			},
 		}
+		if fc.Labels == nil {
+			fc.Labels = map[string]string{}
+		}
+		fc.Labels["searchdomain"] = sd.Name
```

Comment:
can you parameterize this label key? (I would suggest `discovery.liqo.io/searchdomain`)